canonical-web-and-design/vanilla-framework
# Theming correction

## Done

- Restore ability to override nav background colour with `$color-navigation-background`;
- Convert colours map to a variables for easier overriding

## QA

- Pull code
- Run `./run serve --watch`
- Open examples/patterns/navigation/default/, verify it looks as expected
- Open examples/patterns/navigation/default-dark/, verify it looks as expected
- Run any other website like ubuntu.com with ./run --node-module {PATH_TO_VANILLA}
- In the other website, set `$color-navigation-background` to a value above and a value below the threshold (70) `hsl(350, 100%, {INSERT_VALUE_HERE});`
- Observe text colour switch between dark and light 
- Add is-dark/is-light classes, observe they also work
- inside vanilla's `_settings_themes.scss`, change the value of the key 'nav' from false to true; repeat above steps

## Details

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/2566

## Screenshots
The lightness check switches colours at lightness > 70; shouldn't this be 50?
Screenshots of the color switching:
![image](https://user-images.githubusercontent.com/2741678/66396135-fd151f00-e9d0-11e9-8108-844638bd7769.png)
![image](https://user-images.githubusercontent.com/2741678/66396184-1918c080-e9d1-11e9-9589-b4be0aea1e2c.png)
